### PR TITLE
Add support for borderless window

### PR DESCRIPTION
**Pull request by ramn-v**
Go borderless by removing window decorations using _MOTIF_WM_HINTS. This
option can be enabled in the Xresources file with the 'borderless'
option set to one. The code has been shamelessly copied from
https://stackoverflow.com/questions/1904445/borderless-windows-on-linux.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 *.o
 st
```
Comment (actionless) [inline .gitignore:2]: in that case i guess `st` also need to be removed

Comment (actionless) [inline .gitignore:2]:
```suggestion
```

**Comment by actionless** [comment]
since no one else have any objections, merging this one

**Comment by neeasade** [comment]
need to note somewhere -- this is currently broken in master (we don't load borderless state from xrdb until AFTER xinit, so it doesn't get the chance to affect window creation)

**Pull request merged**